compiler/rustc_mir/src/transform/uninhabited_enum_branching.rs

```diff
@@ -89,7 +91,7 @@ impl<'tcx> MirPass<'tcx> for UninhabitedEnumBranching {
             let layout = tcx.layout_of(tcx.param_env(body.source.def_id()).and(discriminant_ty));
 
             let allowed_variants = if let Ok(layout) = layout {
-                variant_discriminants(&layout, discriminant_ty, tcx)
+                FxHashSet::from_iter(variant_discriminants(&layout, discriminant_ty, tcx))
```

Comment:
```suggestion
                variant_discriminants(&layout, discriminant_ty, tcx).collect()
```

..then you can remove the FromIterator import

Reply:
`variant_discriminants` could be changed to return a `FxHashSet` directly which would avoid some overhead.

Reply:
Addressed in latest commit